com.unity.perception/Runtime/GroundTruth/Labelers/BoundingBox3DLabeler.cs

```diff
@@ -0,0 +1,256 @@
+                        
+                        // Apply the transformations on this object until we reach the labeled transform
+                        while (currentTransform != labelTransform)
+                        {
+                            transformedBounds.center += currentTransform.localPosition;
+                            transformedBounds.extents = Vector3.Scale(transformedBounds.extents, currentTransform.localScale);
+                            transformedRotation *= currentTransform.localRotation;
+                            currentTransform = currentTransform.parent;
+                        }
+
+                        // Due to rotations that may be applied, we cannot simply use the extents of the bounds, but
+                        // need to calculate all 8 corners of the bounds and combine them with the current combined
+                        // bounds
+                        var corners = GetBoxCorners(transformedBounds, transformedRotation);
+                        
+                        // If this is the first time, create a new bounds struct
+                        if (areBoundsUnset)
+                        {
+                            combinedBounds = new Bounds(corners[0], Vector3.zero);
+                            areBoundsUnset = false;
+                        }
+
+                        // Go through each corner add add it to the bounds
+                        foreach (var c2 in corners)
+                        {
+                            combinedBounds.Encapsulate(c2);
+                        }
+                    }
+                    
+                    // Convert the combined bounds into world space
+                    combinedBounds.center = labelTransform.TransformPoint(combinedBounds.center);
+                    combinedBounds.extents = Vector3.Scale(combinedBounds.extents,  labelTransform.localScale);
+
+                    // Now convert all points into camera's space
+                    var cameraCenter = cameraTransform.InverseTransformPoint(combinedBounds.center);
+                    cameraCenter = Vector3.Scale(cameraTransform.localScale, cameraCenter);
+                    
+                    // Rotation to go from label space to camera space
+                    var cameraRotation = Quaternion.Inverse(cameraTransform.rotation) * labelTransform.rotation;
+                    
+                    var converted = ConvertToBoxData(labelEntry, groundTruthInfo.instanceId, cameraCenter, combinedBounds.extents, cameraRotation);
+                    BoundingBoxComputed?.Invoke(m_CurrentFrame, converted);
+                    m_BoundingBoxValues.Add(converted);
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public void OnEndUpdate()
+        {
+            perceptionCamera.SensorHandle.ReportAnnotationAsync(m_AnnotationDefinition).ReportValues(m_BoundingBoxValues);
```

Comment:
I don't see anything for only reporting objects that are visible in the frame, which was part of the v1 design before. Am I missing it?

Reply:
No, you are correct, I completely forgot about that. I will implement that immediately.

Reply:
Ok, this has been fixed